### PR TITLE
fix developer link

### LINKED ISSUE #23441: Link to 'Developer documentation' doesn't work on Apps page
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Upgrade to Nextcloud 20
2. Go to Apps page
3. Click on 'Developer documentation' link at the bottom  of the navigation pane on the left

### Expected behaviour
You should go the developer documentation website (I think https://docs.nextcloud.com/server/20/developer_manual/)

### Actual behaviour
Nothing happens as the `href` for the link is `#`

### Server configuration

**Operating system:** Ubuntu 20.04

**Web server:** Nginx 1.19.3

**Database:** Postgresql 12.4

**PHP version:** 7.4.11

**Nextcloud version:** 20.0.0

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated from 19

**Where did you install Nextcloud from:** -

### Client configuration
**Browser:** Firefox 81.0 and Chromium 85.0.4183.121

**Operating system:** Ubuntu 18.04

### PR DESCRIPTION
**Pull request by szaimen**
Fix #23441

Signed-off-by: szaimen <szaimen@e.mail.de>

<details>
<summary>For my own testing</summary>

```
docker run -it \
-e SERVER_BRANCH=fix/23441/fix-developer-link \
-p 8443:443 \
-e TRUSTED_DOMAIN=192.168.146.128 \
--name nextcloud-easy-test \
ghcr.io/szaimen/nextcloud-easy-test:latest
```

</details>

### CODE REVIEW

File: apps/settings/src/views/Apps.vue [hunk 1]
```diff
@@ -75,7 +75,7 @@
 				</template>
 
 				<AppNavigationItem id="app-developer-docs"
-					href="settings.developerDocumentation"
+					:to="settings.developerDocumentation"
```
Comment (szaimen) [inline apps/settings/src/views/Apps.vue:78]: @skjnldsv how to pass an absolute URL here? Seems like the to only accepts relative urls.
and `:href="settings.developerDocumentation"` unfortunately didn't worked for me...

Comment (skjnldsv) [reply apps/settings/src/views/Apps.vue:78]: https://router.vuejs.org/api/#to

Should work with a standard url though :thinking:

Comment (szaimen) [reply apps/settings/src/views/Apps.vue:78]: https://reactgo.com/vue-open-link-new-tab/

File: apps/settings/src/views/Apps.vue [hunk 2]
```diff
@@ -75,7 +75,7 @@
 				</template>
 
 				<AppNavigationItem id="app-developer-docs"
-					href="settings.developerDocumentation"
+					@click="window.open(settings.developerDocumentation)"
```
Comment (szaimen) [inline apps/settings/src/views/Apps.vue:78]: @skjnldsv this doesn't seem to work either. Do you have some other idea how to fix it?

Comment (skjnldsv) [reply apps/settings/src/views/Apps.vue:78]: You need this in a  method.

Comment (skjnldsv) [reply apps/settings/src/views/Apps.vue:78]: TypeError: Cannot read properties of undefined (reading 'open')
    at click (Apps.vue?vue&type=template&id=511b7e80&scoped=true&:4)
    at Zt (vue.runtime.esm.js:1863)
    at o.n (vue.runtime.esm.js:2188)
    at Zt (vue.runtime.esm.js:1863)
    at o.t.$emit (vue.runtime.esm.js:3903)
    at onClick (AppNavigationItem.js:66)
    at Zt (vue.runtime.esm.js:1863)
    at HTMLAnchorElement.n (vue.runtime.esm.js:2188)
    at HTMLAnchorElement.a._wrapper (vue.runtime.esm.js:6961)

Comment (szaimen) [reply apps/settings/src/views/Apps.vue:78]: Itried it in a method and then eslint said:
`settings is undefined`. How can I solve this?

Comment (skjnldsv) [reply apps/settings/src/views/Apps.vue]: Done

**Review by skjnldsv — APPROVED** [review]

File: apps/settings/src/views/Apps.vue (hunk 2 above)
Comment (szaimen) [reply apps/settings/src/views/Apps.vue]: seems like I missed `this` before settings.
Thanks for fixing it!

Comment (szaimen) [reply apps/settings/src/views/Apps.vue]: tested and works 👍

**Review by artonge — APPROVED** [review]

**Comment by szaimen** [comment]
Seems like it really needs to be done manuall

**Pull request merged**